src/ripple/app/tests/SHAMapStore_test.cpp

```diff
@@ -0,0 +1,575 @@
+        section.set("online_delete", to_string(deleteInterval));
+        //section.set("age_threshold", "60");
+        return p;
+    }
+
+    static
+    std::unique_ptr<Config>
+    makeConfigAdvisory()
+    {
+        auto p = makeConfig();
+        auto& section = p->section(ConfigSection::nodeDatabase());
+        section.set("advisory_delete", "1");
+        return p;
+    }
+
+    bool goodLedger(Json::Value const& json, std::string ledgerID)
+    {
+        return json.isMember(jss::result)
+            && !RPC::contains_error(json[jss::result])
+            && json[jss::result][jss::ledger][jss::ledger_index] == ledgerID;
+    }
+
+    bool bad(Json::Value const& json, error_code_i error = rpcLGR_NOT_FOUND)
+    {
+        return json.isMember(jss::result)
+            && RPC::contains_error(json[jss::result])
+            && json[jss::result][jss::error_code] == error;
+    }
+
+    std::string getHash(Json::Value const& json)
+    {
+        expect(json.isMember(jss::result) &&
+            json[jss::result].isMember(jss::ledger) &&
+            json[jss::result][jss::ledger].isMember(jss::hash) &&
+            json[jss::result][jss::ledger][jss::hash].isString());
+        return json[jss::result][jss::ledger][jss::hash].asString();
+    }
+
+    void validationCheck(jtx::Env& env, int const expected)
+    {
+        auto db = env.app().getLedgerDB().checkoutDb();
+
+        int actual;
+        *db << "SELECT count(*) AS rows FROM Validations;",
+            soci::into(actual);
+
+        expect(actual == expected);
+
+    }
+
+    void ledgerCheck(jtx::Env& env, int const rows,
+        int const first)
+    {
+        auto db = env.app().getLedgerDB().checkoutDb();
+
+        int actualRows, actualFirst, actualLast;
+        *db << "SELECT count(*) AS rows, "
+            "min(LedgerSeq) as first, "
+            "max(LedgerSeq) as last "
+            "FROM Ledgers;",
+            soci::into(actualRows),
+            soci::into(actualFirst),
+            soci::into(actualLast);
+
+        expect(actualRows == rows);
+        expect(actualFirst == first);
+        expect(actualLast == first + rows - 1);
+
+    }
+
+    void transactionCheck(jtx::Env& env, int const rows)
+    {
+        auto db = env.app().getTxnDB().checkoutDb();
+
+        int actualRows;
+        *db << "SELECT count(*) AS rows "
+            "FROM Transactions;",
+            soci::into(actualRows);
+
+        expect(actualRows == rows);
+    }
+
+    void accountTransactionCheck(jtx::Env& env, int const rows)
+    {
+        auto db = env.app().getTxnDB().checkoutDb();
+
+        int actualRows;
+        *db << "SELECT count(*) AS rows "
+            "FROM AccountTransactions;",
+            soci::into(actualRows);
+
+        expect(actualRows == rows);
+    }
+
+    int waitForReady(jtx::Env& env)
+    {
+        using namespace std::chrono_literals;
+
+        auto store = dynamic_cast<SHAMapStoreImp*>(
+            &env.app().getSHAMapStore());
```

Comment:
Don't need a `dynamic_cast` here like you do below. `auto& store = env.app().getSHAMapStore()` should work.